File: app/src/main/java/com/citsadigital/relojmultifuncionalcl/custompreference/ListPreferenceAutoSummary.java
package com.citsadigital.relojmultifuncionalcl.custompreference;

import android.content.Context;
import android.util.AttributeSet;

/**
 * Created by Joaquín Alan Alvidrez Soto on 14/03/2018.
 */

public class ListPreferenceAutoSummary extends android.support.v7.preference.ListPreference {
    public ListPreferenceAutoSummary(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ListPreferenceAutoSummary(Context context) {
        super(context);
    }

    @Override
    public void setValue(String value) {
        super.setValue(value);
        setSummary(value);
    }

    @Override
    public void setSummary(CharSequence summary) {
        super.setSummary(getEntry());
    }


}

File: app/src/main/java/com/citsadigital/relojmultifuncionalcl/custompreference/ListPreferenceAutoSummary.kt
package com.citsadigital.relojmultifuncionalcl.custompreference

import android.content.Context
import android.util.AttributeSet

/**
 * Created by Joaquín Alan Alvidrez Soto on 14/03/2018.
 */

class ListPreferenceAutoSummary : android.support.v7.preference.ListPreference {
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context) : super(context)

    override fun setValue(value: String) {
        super.setValue(value)
        summary = value
    }

    override fun setSummary(summary: CharSequence) {
        super.setSummary(entry)
    }


}
